packages/app-builder-lib/src/electron/electronMac.ts

```diff
@@ -69,38 +70,26 @@ export async function createMacApp(packager: MacPackager, appOutDir: string, asa
   const helperGPUPlistFilename = path.join(frameworksPath, `${electronBranding.productName} Helper (GPU).app`, "Contents", "Info.plist")
   const helperLoginPlistFilename = path.join(loginItemPath, `${electronBranding.productName} Login Helper.app`, "Contents", "Info.plist")
 
-  const plistContent: Array<any> = await executeAppBuilderAsJson([
-    "decode-plist",
-    "-f",
-    appPlistFilename,
-    "-f",
-    helperPlistFilename,
-    "-f",
-    helperEHPlistFilename,
-    "-f",
-    helperNPPlistFilename,
-    "-f",
-    helperRendererPlistFilename,
-    "-f",
-    helperPluginPlistFilename,
-    "-f",
-    helperGPUPlistFilename,
-    "-f",
-    helperLoginPlistFilename,
-  ])
+  const safeParsePlistFile = async (filePath: string): Promise<PlistValue | null> => {
+    if (!fs.existsSync(filePath)) {
+      return null
+    }
+    return await parsePlistFile(filePath)
+  }
 
-  if (plistContent[0] == null) {
+  const appPlist = (await safeParsePlistFile(appPlistFilename)) as PlistObject
+  if (appPlist == null) {
     throw new Error("corrupted Electron dist")
   }
 
-  const appPlist = plistContent[0]!
-  const helperPlist = plistContent[1]!
-  const helperEHPlist = plistContent[2]
-  const helperNPPlist = plistContent[3]
-  const helperRendererPlist = plistContent[4]
-  const helperPluginPlist = plistContent[5]
-  const helperGPUPlist = plistContent[6]
-  const helperLoginPlist = plistContent[7]
+  // Replace the multiple parsePlistFile calls with:
+  const helperPlist = (await safeParsePlistFile(helperPlistFilename)) as PlistObject
+  const helperEHPlist = (await safeParsePlistFile(helperEHPlistFilename)) as string | null
+  const helperNPPlist = (await safeParsePlistFile(helperNPPlistFilename)) as string | null
+  const helperRendererPlist = (await safeParsePlistFile(helperRendererPlistFilename)) as string | null
+  const helperPluginPlist = (await safeParsePlistFile(helperPluginPlistFilename)) as string | null
+  const helperGPUPlist = (await safeParsePlistFile(helperGPUPlistFilename)) as string | null
+  const helperLoginPlist = (await safeParsePlistFile(helperLoginPlistFilename)) as PlistObject | null
```

Comment:
I'm really not a fan of this casting. If you make the function generic, you won't need to worry about casting everything to `string` or `PlistObject`. I'm also not sure I understand the logic here. Why are some `string` and others `PlistObject`. Shouldn't they all be returning the same type?

```
export async function parsePlistFile<T>(file: string): Promise<T> {

```

Reply:
>I'm also not sure I understand the logic here. Why are some string and others PlistObject

This is defined based on how the parsed results are intended to be used. For `PlistObject`, properties are directly assigned as shown below. 
https://github.com/electron-userland/electron-builder/blob/f1b519e5671d6e49fcbbd5641d163916c45fb33d/packages/app-builder-lib/src/electron/electronMac.ts#L126-L129


Some of them are strings, as illustrated.
https://github.com/electron-userland/electron-builder/blob/f1b519e5671d6e49fcbbd5641d163916c45fb33d/packages/app-builder-lib/src/electron/electronMac.ts#L25-L30

Reply:
```
export async function parsePlistFile<T>(file: string): Promise<T> {
```
Great suggestion, it has been updated accordingly.

Reply:
>All we're testing in getAvailableHelperSuffixes is whether they're null or not though. AFAICT, parse is always supposed to return an object, not a string https://www.npmjs.com/package/plist

Yes, I checked the code, and it indeed only validates whether it is `null`, so it shouldn't be a `string`. I've updated all instances.